pkg/controller/database/cockroach/cockroach.go

```diff
@@ -90,14 +90,14 @@ func (c *cockroachConnecter) Connect(ctx context.Context, mg resource.Managed) (
 	}
```

Comment:
^ You'll need to change how the provider secret is acquired at line 87.

```diff
- n = types.NamespacedName{Namespace: p.Namespace, Name: p.Spec.Secret.Name}
+ n= types.NamespacedName{Namespace: p.Spec.Secret.Namespace, Name: p.Spec.Secret.Name}
```

Reply:
I wish cluster-scoped CRD's excluded `namespace` from metadata so this would show as error :man_shrugging: